cluster/src/main/java/com/hartwig/pipeline/calling/sage/SageCaller.java

```diff
@@ -0,0 +1,101 @@
+
+    private final PersistedDataset persistedDataset;
+    private final DataType dataType;
+
+    public SageCaller(final AlignmentPair alignmentPair, final PersistedDataset persistedDataset, final DataType dataType) {
+        super(alignmentPair);
+        this.persistedDataset = persistedDataset;
+        this.dataType = dataType;
+    }
+
+    protected abstract String filteredOutput(final SomaticRunMetadata metadata);
+
+    protected abstract String unfilteredOutput(final SomaticRunMetadata metadata);
+
+    @Override
+    public VirtualMachineJobDefinition vmDefinition(final BashStartupScript bash, final ResultsDirectory resultsDirectory) {
+        return VirtualMachineJobDefinition.sageSomaticCalling(bash, resultsDirectory);
+    }
+
+    @Override
+    public SageOutput output(final SomaticRunMetadata metadata, final PipelineStatus jobStatus, final RuntimeBucket bucket,
+            final ResultsDirectory resultsDirectory) {
+        return outputBuilder(metadata, jobStatus, bucket, resultsDirectory).build();
+    }
+
+    protected ImmutableSageOutput.Builder outputBuilder(final SomaticRunMetadata metadata, final PipelineStatus jobStatus,
+            final RuntimeBucket bucket, final ResultsDirectory resultsDirectory) {
+
+        final String filteredOutputFile = filteredOutput(metadata);
+        final String unfilteredOutputFile = unfilteredOutput(metadata);
+
+        return SageOutput.builder(namespace())
+                .status(jobStatus)
+                .addFailedLogLocations(GoogleStorageLocation.of(bucket.name(), RunLogComponent.LOG_FILE))
+                .maybeFinalVcf(GoogleStorageLocation.of(bucket.name(), resultsDirectory.path(filteredOutputFile)))
+                .addReportComponents(bqrComponent(metadata.tumor(), "png", bucket, resultsDirectory))
+                .addReportComponents(bqrComponent(metadata.tumor(), "tsv", bucket, resultsDirectory))
+                .addReportComponents(bqrComponent(metadata.reference(), "png", bucket, resultsDirectory))
+                .addReportComponents(bqrComponent(metadata.reference(), "tsv", bucket, resultsDirectory))
+                .addReportComponents(vcfComponent(unfilteredOutputFile, bucket, resultsDirectory))
+                .addReportComponents(vcfComponent(filteredOutputFile, bucket, resultsDirectory))
+                .addReportComponents(new RunLogComponent(bucket, namespace(), Folder.root(), resultsDirectory))
+                .addReportComponents(new StartupScriptComponent(bucket, namespace(), Folder.root()))
+                .addFurtherOperations(new AddDatatypeToFile(dataType, Folder.root(), namespace(), filteredOutputFile, metadata.barcode()));
+    }
+
+    @Override
+    public SageOutput skippedOutput(final SomaticRunMetadata metadata) {
+        return SageOutput.builder(namespace()).status(PipelineStatus.SKIPPED).build();
+    }
+
+    @Override
+    public final SageOutput persistedOutput(final SomaticRunMetadata metadata) {
```

Comment:
Maybe this guy is better only to implement in the somatic caller. Generally "persisted output" is only required for stages with other stages which depend on them, so you could leave this out of sage germline.

Reply:
Left as discussed in email thread as we will be running GATK and sage germline in parallel for a while.